internal/rbd/rbd_util.go

```
@@ -1918,3 +1929,40 @@ func genVolFromVolIDWithMigration(
 
 	return rv, err
 }
+
+// Set PV/PVC/PVCNamespace metadata on RBD image.
+func (rv *rbdVolume) setVolumeMetadata(
+	ctx context.Context, parameters map[string]string) error {
+	s := []string{pvcNameKey, pvcNamespaceKey, pvNameKey}
+	for _, i := range s {
+		if parameters[i] != "" {
+			err := rv.SetMetadata(i, parameters[i])
+			if err != nil {
+				log.ErrorLog(ctx, "failed to set metadata key %q, value %q on image: %v", i, parameters[i], err)
```

instead of logging, just return the error with the additional context?